File: ruby/if/if.rb
# encoding: UTF-8
require 'rubygems'
require 'awesome_print'

@a = ['1','2']
def slug_exists slug
  @a.include? slug
end
slug = '1'
new_slug = slug
i = 1

while slug_exists(new_slug)
  i+= 1
  new_slug = "#{slug}-#{i}"
end

print 'new_slug: '.red; puts new_slug

puts 'Если значение поппадает в диапазон.'.green
a = 2
if 1 > a || a > 3
  puts "bad"
else
  puts "good"
end
if 1 <= a && a <= 3
  puts "good"
else
  puts "bad"
end

File: python/if/if.py
# -*- coding: utf-8 -*-
from termcolor import colored

print(colored('Полный синтаксис if', 'green'))
if True:
  print(True)
elif True:
  print(True)
else:
  print(True)

print(colored('if и None', 'green'))
if None:
  print('None is not true.')
if None is not None:
  print('None is not None.')
if None is None:
  print('None Is None.')

print(colored('Булевы операции в if', 'green'))
if True & (False | True):
  print(True)

a = True if True else False
print("a:", a)
